presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -505,9 +507,37 @@ public Filter getMember(SubfieldPath.PathElement member)
             return filters.get(member);
         }
 
+        public Filter getMember(long subscript)
+        {
+            return longToFilter.get(subscript);
+        }
+
         public void addMember(SubfieldPath.PathElement member, Filter filter)
         {
             filters.put(member, filter);
+            if (member.getField() == null) {
+                if (longToFilter == null) {
+                    longToFilter = new HashMap();
+                }
+                longToFilter.put(Long.valueOf(member.getSubscript()), filter);
+            }
+        }
+        public HashMap<SubfieldPath.PathElement, Filter> getFilters()
```

Comment:
Changes to StructFilter don't belong to this PR. Remove them. (This method should return Map, not HashMap.)